tests/Feature/Actions/StoreCustomFieldsTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace TestsNextGen\Feature\Actions;
+
+use Give\Donations\Models\Donation;
+use Give\Framework\Database\DB;
+use Give\Framework\Exceptions\Primitives\Exception;
+use Give\NextGen\DonationForm\Actions\StoreCustomFields;
+use Give\NextGen\DonationForm\Models\DonationForm;
+use Give\NextGen\Framework\Blocks\BlockCollection;
+use Give\NextGen\Framework\Blocks\BlockModel;
+use GiveTests\TestCase;
+use GiveTests\TestTraits\RefreshDatabase;
+
+class StoreCustomFieldsTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * @unreleased
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function testShouldStoreAsDonorMeta()
+    {
+          /** @var DonationForm $form */
+        $form = DonationForm::factory()->create();
+
+        $customFieldBlockModel =  BlockModel::make([
+            'name' => 'custom-block-editor/section',
+            'attributes' => [ 'title' => '', 'description' => '' ],
+            'innerBlocks' => [
+                [
+                    'name' => 'custom-block-editor/custom-text-block',
+                    'attributes' => [
+                        'fieldName' => 'custom_text_block_meta',
+                        'storeAsDonorMeta' => true,
+                        'title' => 'Custom Text Field',
+                        'description' => ''
+                    ],
+                ]
+            ]
+        ]);
+
+        $form->blocks = BlockCollection::make(
+            array_merge([$customFieldBlockModel], $form->blocks->getBlocks())
+        );
+
+        $form->save();
+
+        /** @var Donation $donation */
+        $donation = Donation::factory()->create(['formId' => $form->id]);
+
+        $action = new StoreCustomFields();
+
+        $action($form, $donation, ['custom_text_block_meta' => 'Custom Text Block Value']);
+
+        $query = DB::table('give_donormeta')
+            ->select('meta_value')
+            ->where('donor_id', $donation->donorId)
+            ->where('meta_key', 'custom_text_block_meta')
+        ->get();
```

Comment:
I'm seeing random, super small, formatting things. Make sure you have "Reformat on Save" turned on. 😄

Reply:
idk why but php storm reformat on save doesn't always work for me - even with that setting on. It's something I need to look more into it.  I often have to manually format when I notice it not hapenning.  8e27569e